graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -537,13 +580,13 @@ def _set_up_schema_elements_of_kind(self, class_name_to_definition, kind, class_
                         property_name_to_descriptor[property_name] = property_descriptor
 
                 if (property_name not in property_name_to_descriptor and not abstract and
-                        kind == SchemaElement.ELEMENT_KIND_EDGE):
+                        kind == Edge):
                     raise AssertionError(u'For property "{}" of non-abstract edge class "{}", '
                                          u'no such subclass-of-all-elements exists.'
                                          .format(property_name, class_name))
 
-            self._elements[class_name] = SchemaElement(class_name, kind, abstract,
-                                                       property_name_to_descriptor, class_fields)
+            self._elements[class_name] = kind(class_name, abstract, property_name_to_descriptor,
```

Comment:
If you are going to call it as a constructor, call it `element_cls` or something like that, to make it obvious that it's a type and it can be called as a constructor.